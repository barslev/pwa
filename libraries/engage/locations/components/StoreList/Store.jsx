import React, { useContext } from 'react';
import StoreContext from './Store.context';
import StoreAddress from './StoreAddress';

/**
 * Renders the store's addresses as pickup locations.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function Store() {
  const store = useContext(StoreContext);

  if (!store || !store.addresses || store.addresses.length === 0) {
    return null;
  }

  return store.addresses.map(address => (
    <StoreAddress address={address} key={address.code} />
  ));
}

export default Store;
