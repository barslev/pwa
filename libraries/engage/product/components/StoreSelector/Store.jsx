import React, { memo, useContext } from 'react';
import PropTypes from 'prop-types';
import { Accordion } from '@shopgate/pwa-ui-material';
import I18n from '@shopgate/pwa-common/components/I18n';
import Address from './Address';
import StoreHoursToday from './StoreHoursToday';
import PhoneNumber from './PhoneNumber';
import OpeningHours from './OpeningHours';
import {
  store as storeStyles, storeMain, storeImage, storeName, storeInfo,
} from './style';
import StoreSelectorContext from './context';
import LocationStockInfo from '../LocationStockInfo';

/**
 * Renders the store's addresses as pickup locations.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const Store = ({ store }) => {
  const { selectLocation } = useContext(StoreSelectorContext);

  if (!store || !store.addresses || store.addresses.length === 0) {
    return null;
  }

  return store.addresses.map((address) => {
    /**
     * Handles the click action.
     */
    const handleClick = () => {
      selectLocation({
        code: store.code,
        name: store.name,
        addressCode: address.code,
        visibleInventory: store.inventory.visible,
      });
    };

    return (
      <div className={storeStyles} key={address.code}>
        <div className={storeMain}>
          <div className={storeImage} />
          <div
            className={storeInfo}
            onClick={handleClick}
            onKeyDown={handleClick}
            role="button"
            tabIndex={0}
          >
            <div className={storeName} data-test-id="store-name">{store.name}</div>
            <StoreHoursToday hours={store.operationHours} />
            <Address address={address} />
            <LocationStockInfo
              location={{
                storeName: store.name,
                visibleInventory: store.inventory.visible,
              }}
              showStoreName={false}
            />
          </div>
        </div>
        <Accordion renderLabel={() => <I18n.Text string="product.location.details" />}>
          <OpeningHours hours={store.operationHours} />
          <PhoneNumber phone={address.phoneNumber} />
        </Accordion>
      </div>
    );
  });
};

Store.propTypes = {
  store: PropTypes.shape(),
};

Store.defaultProps = {
  store: null,
};

export default memo(Store);
