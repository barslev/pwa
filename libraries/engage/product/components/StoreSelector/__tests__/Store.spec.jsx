import React, { useContext } from 'react';
import { shallow } from 'enzyme';
import Store from '../Store';

const context = {
  selectLocation: jest.fn(),
};

const store = {
  code: 'store',
  name: 'SomeStore',
  operationHours: {
    mon: '8:00am - 4:00pm',
  },
  inventory: {
    visible: 10,
  },
  addresses: [
    {
      code: 'address',
      phoneNumber: '000-000-0000',
    },
  ],
};

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest.fn(),
}));

describe('<Store />', () => {
  it('should not render if no store is passed', () => {
    useContext.mockReturnValueOnce(context);
    const wrapper = shallow(<Store />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should render as expected', () => {
    useContext.mockReturnValueOnce(context);
    const wrapper = shallow(<Store store={store} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('[data-test-id="store-name"]').text()).toEqual(store.name);
    expect(wrapper.find('StoreHoursToday').props().hours).toEqual(store.operationHours);
    expect(wrapper.find('OpeningHours').props().hours).toEqual(store.operationHours);
    expect(wrapper.find('Address').props().address).toEqual(store.addresses[0]);
    expect(wrapper.find('PhoneNumber').props().phone).toEqual(store.addresses[0].phoneNumber);
  });

  it('should handle the store selection', () => {
    useContext.mockReturnValueOnce(context);
    const wrapper = shallow(<Store store={store} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('div[role="button"]').simulate('click');
    expect(context.selectLocation).toHaveBeenCalledWith({
      code: store.code,
      name: store.name,
      addressCode: store.addresses[0].code,
      visibleInventory: store.inventory.visible,
    });
  });
});
