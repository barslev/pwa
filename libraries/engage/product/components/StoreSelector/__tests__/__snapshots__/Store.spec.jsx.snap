// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Store /> should handle the store selection 1`] = `
<div
  className={
    Object {
      "data-css-n58wjh": "",
    }
  }
  key="address"
>
  <div
    className={
      Object {
        "data-css-1ixep8k": "",
      }
    }
  >
    <div
      className={
        Object {
          "data-css-10jbyiq": "",
        }
      }
    />
    <div
      className={
        Object {
          "data-css-rw5zfw": "",
        }
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className={
          Object {
            "data-css-1qlb3ci": "",
          }
        }
        data-test-id="store-name"
      >
        SomeStore
      </div>
      <Component
        hours={
          Object {
            "mon": "8:00am - 4:00pm",
          }
        }
      />
      <Component
        address={
          Object {
            "code": "address",
            "phoneNumber": "000-000-0000",
          }
        }
      />
      <Component
        location={
          Object {
            "storeName": "SomeStore",
            "visibleInventory": 10,
          }
        }
        showStoreName={false}
      />
    </div>
  </div>
  <Accordion
    handleLabel={null}
    renderLabel={[Function]}
    testId={null}
  >
    <Component
      hours={
        Object {
          "mon": "8:00am - 4:00pm",
        }
      }
    />
    <Component
      phone="000-000-0000"
    />
  </Accordion>
</div>
`;

exports[`<Store /> should not render if no store is passed 1`] = `""`;

exports[`<Store /> should render as expected 1`] = `
<div
  className={
    Object {
      "data-css-n58wjh": "",
    }
  }
  key="address"
>
  <div
    className={
      Object {
        "data-css-1ixep8k": "",
      }
    }
  >
    <div
      className={
        Object {
          "data-css-10jbyiq": "",
        }
      }
    />
    <div
      className={
        Object {
          "data-css-rw5zfw": "",
        }
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className={
          Object {
            "data-css-1qlb3ci": "",
          }
        }
        data-test-id="store-name"
      >
        SomeStore
      </div>
      <Component
        hours={
          Object {
            "mon": "8:00am - 4:00pm",
          }
        }
      />
      <Component
        address={
          Object {
            "code": "address",
            "phoneNumber": "000-000-0000",
          }
        }
      />
      <Component
        location={
          Object {
            "storeName": "SomeStore",
            "visibleInventory": 10,
          }
        }
        showStoreName={false}
      />
    </div>
  </div>
  <Accordion
    handleLabel={null}
    renderLabel={[Function]}
    testId={null}
  >
    <Component
      hours={
        Object {
          "mon": "8:00am - 4:00pm",
        }
      }
    />
    <Component
      phone="000-000-0000"
    />
  </Accordion>
</div>
`;
